quickjs/src/main/jni/quickjs-jni.c

```diff
@@ -56,22 +54,34 @@ Java_com_squareup_quickjs_QuickJs_evaluate__JLjava_lang_String_2Ljava_lang_Strin
 
   JSValue evalValue = JS_Eval(context, sourceCode, strlen(sourceCode), fileName, 0);
 
+  (*env)->ReleaseStringUTFChars(env, sourceCode_, sourceCode);
+  (*env)->ReleaseStringUTFChars(env, fileName_, fileName);
+
   jstring result;
   switch (JS_VALUE_GET_TAG(evalValue)) {
     case JS_TAG_EXCEPTION: {
-      JS_FreeValue(context, evalValue);
-
       JSValue exceptionValue = JS_GetException(context);
 
       JSValue messageValue = JS_GetPropertyStr(context, exceptionValue, "message");
+      JSValue stackValue = JS_GetPropertyStr(context, exceptionValue, "stack");
       JS_FreeValue(context, exceptionValue);
 
       const char *message = JS_ToCString(context, messageValue);
       JS_FreeValue(context, messageValue);
 
+      const char *stack = JS_ToCString(context, stackValue);
+      JS_FreeValue(context, stackValue);
+
       jclass Exception = (*env)->FindClass(env, "com/squareup/quickjs/QuickJsException");
-      (*env)->ThrowNew(env, Exception, message);
+      char *messageAndStack = (char*) malloc(strlen(message) + strlen(stack) + 2);
+      strcpy(messageAndStack, message);
+      strcat(messageAndStack, "\n");
```

Comment:
Thinking that we should just create a two-argument constructor instead of doing a concatenation only to tear it back apart. Can do in a follow-up.

Reply:
Good call, this does feel a little clunky.